utils/src/lib.rs

```diff
@@ -1024,23 +1028,35 @@ mod tests {
         let contracts = output
             .into_artifacts()
             .filter(|(i, _)| contract_names.contains(&i.slug().as_str()))
-            .map(|(i, c)| (i.slug(), c.into_contract_bytecode()))
-            .collect::<BTreeMap<String, CompactContractBytecode>>();
+            .map(|(id, c)| (id, c.into_contract_bytecode()))
+            .collect::<BTreeMap<ArtifactId, CompactContractBytecode>>();
 
-        let mut known_contracts: BTreeMap<String, (Abi, Vec<u8>)> = Default::default();
+        let mut known_contracts: BTreeMap<ArtifactId, (Abi, Vec<u8>)> = Default::default();
         let mut deployable_contracts: BTreeMap<String, (Abi, Bytes, Vec<Bytes>)> =
             Default::default();
-        assert_eq!(&contracts.keys().collect::<Vec<&String>>()[..], &contract_names[..]);
+        assert_eq!(
+            &contracts.keys().map(|i| i.slug()).collect::<Vec<String>>()[..],
+            &contract_names[..]
+        );
+
         let lib_linked = hex::encode(
-            &contracts["Lib.json:Lib"]
+            &contracts
+                .iter()
+                .find(|(i, _)| i.slug() == "Lib.json:Lib")
+                .unwrap()
+                .1
                 .bytecode
                 .clone()
                 .expect("library had no bytecode")
                 .object
                 .into_bytes()
                 .expect("could not get bytecode as bytes"),
```

Comment:
![image](https://user-images.githubusercontent.com/8862627/160258567-65d2281e-ce9b-408b-92cd-80b0b3c3e2aa.png)

Reply:
eh this is fine